File: backend/tours/src/main/java/uns/ac/rs/soa/tours/model/Tour.java
package uns.ac.rs.soa.tours.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document("tours")
public class Tour {
    @Id
    private String id;
    private String guideId;
    private String name;
    private String description;
    private String difficulty;
    private List<String> tags;
    private String status;
    private double price;
    private LocalDateTime createdAt;
}

File: frontend/tourapp/src/types/tour.ts
export interface Tour {
  id: string;
  guideId: string;
  name: string;
  description: string;
  difficulty: string;
  tags: string[];
  status: string;
  price: number;
  createdAt: string;
}

export interface Keypoint {
  id: string;
  tourId: string;
  name: string;
  description: string;
  latitude: number;
  longitude: number;
  image: string | null;
  orderIndex: number;
}

export interface CreateTourRequest {
  name: string;
  description: string;
  difficulty: string;
  tags: string[];
  price?: number;
}

export interface UpdateTourRequest {
  name?: string;
  description?: string;
  difficulty?: string;
  tags?: string[];
  status?: string;
  price?: number;
}

export interface CreateKeypointRequest {
  name: string;
  description: string;
  latitude: number;
  longitude: number;
  image?: string;
  orderIndex: number;
}
